std/format.d

```diff
@@ -7538,3 +7553,782 @@ if (is(T == float) || is(T == double) || (is(T == real) && T.mant_dig == double.
     assert(printFloat(0x1.19f81p0, f, FloatingPointControl.roundToNearest) == "0X1.1A0P+0");
     assert(printFloat(0x1.19f01p0, f, FloatingPointControl.roundToNearest) == "0X1.19FP+0");
 }
+
+private auto printFloatE(T, Char)(T val, FormatSpec!Char f, FloatingPointControl.RoundingMode rm,
+                                  string sgn, int exp, ulong mnt, bool is_upper) pure @safe
+if (is(T == float) || is(T == double) || (is(T == real) && T.mant_dig == double.mant_dig))
+{
+    import std.conv : to;
+    import std.algorithm.comparison : max;
+
+    enum int bias = T.max_exp - 1;
+
+    if (f.precision == f.UNSPECIFIED)
+        f.precision = 6;
+
+    // special treatment for 0.0
+    if (exp == 0 && mnt == 0)
+    {
+        char[] result;
+        result.length = max(f.width, f.precision + ((f.precision == 0 && !f.flHash) ? 1 : 2) + sgn.length + 4);
+        result[] = '0';
+
+        if (f.flDash)
+        {
+            if (sgn!="")
+                result[0] = sgn[0];
+
+            int dot_pos = cast(int) (sgn.length + 1);
+            if (f.precision > 0 || f.flHash)
+                result[dot_pos] = '.';
+            if (dot_pos + ((f.precision > 0 || f.flHash) ? 1 : 0) + f.precision + 4 < result.length)
```

Comment:
Please give a name to the expression `dot_pos + ((f.precision > 0 || f.flHash) ? 1 : 0)`